ostelco-ios-client/ViewControllers/Home/BecomeAMemberViewController.swift

```diff
@@ -15,54 +15,81 @@ class BecomeAMemberViewController: ApplePayViewController {
-        // 1. Make sure device supports apple pay and that there are no other restrictions preventing payment (like parental control)
-        if PKPaymentAuthorizationViewController.canMakePayments() {
-            
-            let paymentButton: PKPaymentButton
-            
-            // 2. Check if user has a stripe supported card in its wallet
-            if Stripe.deviceSupportsApplePay() {
-                paymentButton = PKPaymentButton(paymentButtonType: .checkout, paymentButtonStyle: .whiteOutline)
-                paymentButton.addTarget(self, action: #selector(BecomeAMemberViewController.buyButtonTapped), for: .touchUpInside)
-            } else {
-                paymentButton = PKPaymentButton(paymentButtonType: .setUp, paymentButtonStyle: .whiteOutline)
-                paymentButton.addTarget(self, action: #selector(BecomeAMemberViewController.setUpButtonTapped), for: .touchUpInside)
-            }
-            paymentButton.translatesAutoresizingMaskIntoConstraints = false
-            buttonContainer.addSubview(paymentButton)
-            
-            paymentButton.widthAnchor.constraint(equalTo: buttonContainer.widthAnchor).isActive = true
-            paymentButton.heightAnchor.constraint(equalTo: buttonContainer.heightAnchor).isActive = true
-            paymentButton.centerXAnchor.constraint(equalTo: buttonContainer.centerXAnchor).isActive = true
-            paymentButton.centerYAnchor.constraint(equalTo: buttonContainer.centerYAnchor).isActive = true
-            if #available(iOS 12.0, *) {
-                paymentButton.cornerRadius = 8.0
-            }
+        setupPaymentButton()
+        let spinnerView: UIView = showSpinner(onView: view)
+        getProducts { products, error in
+            self.removeSpinner(spinnerView)
+            self.plan = self.getFirstPlan(products)
+        }
+    }
+
+    private func getFirstPlan(_ products: [Product]) -> Product? {
+        // See if the list contains offers.
+        if let firstPlan = products.first(where: { $0.type == "plan" }) {
+            return firstPlan
         }
-        
+        return nil
     }
-    
+
+    func setupPaymentButton() {
+        var showSetupButton = false
+        let applePayError:ApplePayError? = canMakePayments()
+        switch applePayError {
+        case .unsupportedDevice?:
+            debugPrint("Apple Pay is not supported on this device")
+            return
+        case .noSupportedCards?,
+             .otherRestrictions?:
+            showSetupButton = true
+        default:
+            showSetupButton = false
+        }
+        // Properties to Setup Apple Pay.
+        var action = #selector(BecomeAMemberViewController.setUpButtonTapped)
+        var paymentButtonType: PKPaymentButtonType =  .setUp
+        let paymentButton: PKPaymentButton
+        if showSetupButton == false {
+            action = #selector(BecomeAMemberViewController.buyButtonTapped)
+            paymentButtonType =  .checkout
+        }
```

Comment:
this is a little hard to reason about, particularly the `if showSetupButton == false` block.

You can set up the `action` and `paymentButtonType` as `let`s above the `if`/`else` statement (since they never change once assigned) and then assign them within the `if`/`else` statement before using them below. That way they can still be non-optional but it's a lot easier to parse visually when each case is happening.